server/utils/auth/enforcer.py

```
@@ -175,55 +166,63 @@ def get_enforcer() -> casbin.Enforcer:
-    in-process enforcer cache stays current.
 
-    Thread-safe: acquires _lock so concurrent reloads cannot corrupt the
-    enforcer's in-memory policy cache.
-    """
-    with _lock:
-        enforcer = get_enforcer()
-        enforcer.load_policy()
-        logger.info("Casbin policies reloaded from database.")
+def increment_policy_version() -> None:
+    """Bump the Redis policy version counter so other workers know to reload."""
+    global _cached_version
```

Ruff (0.15.15)

[warning] Using the global statement to update `_cached_version` is discouraged

(PLW0603)